src/uri/port.rs

```diff
@@ -0,0 +1,67 @@
+use std::{fmt, num};
+use std::str::FromStr;
+
+/// The port component of a URI.
+#[derive(Debug)]
+pub struct Port<'a> {
+    bytes: &'a str,
+    port: u16,
+}
+
+impl<'a> Port<'a> {
+    /// Converts a `str` to a port number.
+    ///
+    /// The supplied `str` must be a valid u16.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use http::uri::Port;
+    ///
+    /// assert!(Port::from_str("80").is_ok());
+    /// assert!(Port::from_str("abcd").is_err());
+    /// ```
+    pub fn from_str(bytes: &'a str) -> Result<Self, num::ParseIntError> {
```

Comment:
I would not expose this fn yet in favor of not letting the type be publicly constructed yet.